src/test/kotlin/no/skatteetaten/aurora/mokey/service/ManagementEndpointTest.kt

```diff
@@ -0,0 +1,90 @@
+package no.skatteetaten.aurora.mokey.service
+
+import com.jayway.jsonpath.JsonPath
+import no.skatteetaten.aurora.mokey.AbstractTest
+import no.skatteetaten.aurora.mokey.ApplicationConfig
+import no.skatteetaten.aurora.mokey.service.Endpoint.MANAGEMENT
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.junit.jupiter.api.Assertions.assertThrows
+import org.junit.jupiter.api.DynamicTest
+import org.junit.jupiter.api.DynamicTest.dynamicTest
+import org.junit.jupiter.api.TestFactory
+import org.springframework.boot.web.client.RestTemplateBuilder
+import org.springframework.http.HttpStatus
+import org.springframework.http.HttpStatus.*
+import org.springframework.http.MediaType
+import org.springframework.http.MediaType.APPLICATION_JSON
+import org.springframework.http.MediaType.TEXT_HTML
+import org.springframework.test.web.client.ExpectedCount.once
+import org.springframework.test.web.client.MockRestServiceServer
+import org.springframework.test.web.client.match.MockRestRequestMatchers.requestTo
+import org.springframework.test.web.client.response.MockRestResponseCreators.withStatus
+import org.springframework.test.web.client.response.MockRestResponseCreators.withSuccess
+
+class ManagementEndpointTest : AbstractTest() {
+
+    val restTemplate = ApplicationConfig().restTemplate(RestTemplateBuilder())
+    val server = MockRestServiceServer.createServer(restTemplate)
+
+    val managementUrl = "http://localhost:8081/management"
+
+    @Test
+    fun `should use links from management response for health and info endpoints`() {
+
+        val resource = loadResource("management.json")
+        val (healthLink, infoLink) = JsonPath.parse(resource).let { json ->
+            listOf("health", "info").map { json.read("$._links.$it.href", String::class.java) }
+        }
+
+        server.apply {
+            expect(requestTo(managementUrl)).andRespond(withSuccess(resource, APPLICATION_JSON))
+            expect(requestTo(healthLink)).andRespond(withJsonResponse("health.json"))
+            expect(requestTo(infoLink)).andRespond(withJsonResponse("info.json"))
+        }
+
+        val managementEndpoint = ManagementEndpoint.create(restTemplate, managementUrl)
+        managementEndpoint.getHealthEndpointResponse()
+        managementEndpoint.getInfoEndpointResponse()
+    }
+
+    @TestFactory
+    fun `should handle management link response errors`(): Collection<DynamicTest> {
```

Comment:
For meg er denne testen ustabil. Kjører jeg den alene så fungerer den, kjører jeg alle testene så feiler den med denne feilmeldingen:  
`java.lang.IllegalStateException: Cannot add more expectations after actual requests are made`